src/constitutive/TACSGaussianProcessModel.h

```diff
@@ -0,0 +1,162 @@
+  int getNparam() { return n_param; };
+  TacsScalar getKS() { return ks; };
+  void setKS(TacsScalar ks) { this->ks = ks; };
+  void setAlpha(TacsScalar* alpha) { this->alpha = alpha; };
+  void setTheta(TacsScalar* theta) { this->theta = theta; };
+  void getTrainingData(TacsScalar* Xtrain) { Xtrain = this->Xtrain; };
+  void getTheta(TacsScalar* theta) { theta = this->theta; };
+  // virtual functions for the kernel definition and its sensitivity
+  virtual TacsScalar kernel(const TacsScalar* Xtest, const TacsScalar* Xtrain) {
+    return 0.0;
+  };
+
+ protected:
+  virtual void kernelSens(const TacsScalar ksens, const TacsScalar* Xtest,
+                          const TacsScalar* Xtrain, TacsScalar* Xtestsens){};
+
+  int n_train;
+  int n_param;
+  int n_theta = 14;
+  // rank 1-tensor of length [n_param*n_train] [X]
+  // if each point of Xtrain has data [rho0, xi, gamma, delta, zeta] with
+  // n_Train=5 then the entries are basically [rho01, xi1, gamma1, delta1,
+  // zeta1, rho02, xi2, gamma2, delta2, zeta2, ..., zetaN]
+  TacsScalar* Xtrain;
+  TacsScalar* alpha;
+  TacsScalar* theta;  // hyperparameters
+
+  TacsScalar ks = 10.0;  // ks setting for smooth kernel functions
+};
+
+class TACSAxialGaussianProcessModel : public TACSGaussianProcessModel {
+ public:
+  TACSAxialGaussianProcessModel(int n_train, const TacsScalar Xtrain[],
+                                const TacsScalar alpha[],
+                                const TacsScalar theta[])
+      : TACSGaussianProcessModel(n_train, N_PARAM, Xtrain, alpha, theta){};
+  ~TACSAxialGaussianProcessModel(){};
+  TacsScalar testKernelSens(TacsScalar epsilon, int printLevel);
+
+  // here Xtest, Xtrain are each length 5 arrays (N_PARAM) [just uses one train
+  // and one test point here] these are overwritten from subclass
+  TacsScalar kernel(const TacsScalar* Xtest, const TacsScalar* Xtrain);
+
+ protected:
+  void kernelSens(const TacsScalar ksens, const TacsScalar* Xtest,
+                  const TacsScalar* Xtrain, TacsScalar* Xtestsens);
+
+  // there are 4 parameters [log(xi), log(rho_0), log(1+gamma), log(zeta)] for
+  // the axial model
+  static const int N_PARAM = 4;
+};
+
+class TACSShearGaussianProcessModel : public TACSAxialGaussianProcessModel {
+ public:
+  TACSShearGaussianProcessModel(int n_train, const TacsScalar Xtrain[],
+                                const TacsScalar alpha[],
+                                const TacsScalar theta[])
+      : TACSAxialGaussianProcessModel(n_train, Xtrain, alpha, theta){};
+  ~TACSShearGaussianProcessModel(){};
+};
+
+class TACSCripplingGaussianProcessModel : public TACSAxialGaussianProcessModel {
+ public:
+  TACSCripplingGaussianProcessModel(int n_train, const TacsScalar Xtrain[],
+                                    const TacsScalar alpha[],
+                                    const TacsScalar theta[])
+      : TACSAxialGaussianProcessModel(n_train, Xtrain, alpha, theta){};
+  ~TACSCripplingGaussianProcessModel(){};
+};
```

Comment:
That's true, I set it up this way because I was originally testing out different kernel functions for each of them : AxialGP, ShearGP, CripplingGP; especially the CripplingGP. Then it ended up that I just stuck with the same kernel for both the AxialGP and ShearGP and the CripplingGP was less important. I guess I will change it then to make it cleaner